addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/BaseKeypadHandler.java

```diff
@@ -0,0 +1,358 @@
+            this.autoRelease = arParam;
+        }
+
+        // schedule a thread to finish initialization asynchronously since it can take several seconds
+        this.scheduler.schedule(new Runnable() {
+            @Override
+            public void run() {
+                asyncInitialize();
+            }
+        }, 0, TimeUnit.SECONDS);
+    }
+
+    private synchronized void asyncInitialize() {
+        this.logger.debug("Async init thread staring for keypad handler {}", integrationId);
+
+        buttonList = new ArrayList<KeypadComponent>();
+        ledList = new ArrayList<KeypadComponent>();
+        cciList = new ArrayList<KeypadComponent>();
+
+        configureComponents(this.model);
+
+        // load the channel-id map
+        for (KeypadComponent component : buttonList) {
+            ComponentChannelMap.put(component.id(), component.channel());
+        }
+        for (KeypadComponent component : ledList) {
+            ComponentChannelMap.put(component.id(), component.channel());
+        }
+        for (KeypadComponent component : cciList) {
+            ComponentChannelMap.put(component.id(), component.channel());
+        }
+
+        configureChannels();
+
+        updateStatus(ThingStatus.ONLINE);
+
+        this.logger.debug("Async init thread finishing for keypad handler {}", integrationId);
+        return;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, Command command) {
+        logger.debug("Handling command {} for channel {}", command, channelUID);
+
+        Channel channel = getThing().getChannel(channelUID.getId());
+        if (channel == null) {
+            logger.warn("Command received on invalid channel {} for device {}", channelUID,
+                    getThing().getUID().toString());
+            return;
+        }
+
+        Integer componentID = componentFromChannel(channelUID);
+        if (componentID == null) {
+            logger.warn("Command received on invalid channel {} for device {}", channelUID,
+                    getThing().getUID().toString());
+            return;
+        }
+
+        // For LEDs, handle RefreshType and OnOffType commands
+        if (isLed(componentID)) {
+            if (command instanceof RefreshType) {
+                queryDevice(componentID, ACTION_LED_STATE);
+                return;
+            }
+
+            if (command instanceof OnOffType) {
+                if (command == OnOffType.ON) {
+                    device(componentID, ACTION_LED_STATE, LED_ON);
+                } else if (command == OnOffType.OFF) {
+                    device(componentID, ACTION_LED_STATE, LED_OFF);
+                } else {
+                    assert false : "OnOffType command state is neither ON nor OFF";
+                }
+            } else {
+                logger.warn("Invalid command {} received for channel {} device {}", command, channelUID,
+                        getThing().getUID());
+            }
+            return;
```

Comment:
Yes, I agree it could be cleaned up a bit, but at this point I'd rather not mess with it since I don't have the ability to do proper regression testing at the moment.

Reply:
I did simplify this slightly in the commit I'm pushing now.